### DISCUSSION TITLE
Enable search with dot notation - for example: "lodash.map"

### DISCUSSION THREAD

**Discussion by Avivhdr** [community/community#8615] [Other Feature Feedback, Questions, & Ideas] [answered]
In an organization that has hundreds of repositories, there is no way (that I know of) to search using `.`.
The dot notation has a special place of importance due to it being used to access properties, methods, etc in many programming languages.

Let's take the `lodash.map` example. 
If I want to find and deprecate its use in all of our repositories:
Searching `lodash.map` will return 0 results.
Searching `lodash` will return all lodash instances (too wide).
Searching `map` will return `lodash.map` instances and `[array].map` instances (too wide).

To do it locally on my PC I will need to clone hundreds of repositories and then search in the folder that contains them.

Would it be possible to add this feature to the new **Code Search**?
Thank you

**Comment by look** [comment, marked as answer]
This should work in the new code search technology preview: [`lowdash.map language:javascript`](https://cs.github.com/?scopeName=All+repos&scope=&q=lodash.map%20language%3Ajavascript)

**Reply by Avivhdr** [reply]
Nice 🙏🏼
I signed up for the preview.

**Reply by Avivhdr** [reply]
@look I joined yesterday and it works. 
Great 👏🏼

**Reply by LittleEvilBoi** [reply]
Ich bin heute neu dazu beigetreten und bis jetzt finde ich es sehr interessant

**Comment by ghost** [comment]
> ในองค์กรที่มีที่เก็บหลายร้อยแห่ง ไม่มีทาง (ที่ฉันรู้) ที่จะค้นหาโดยใช้`.`. เครื่องหมายจุดมีความสำคัญเป็นพิเศษเนื่องจากใช้เพื่อเข้าถึงคุณสมบัติ วิธีการ ฯลฯ ในภาษาการเขียนโปรแกรมหลายภาษา
> 
> มาดู`lodash.map`ตัวอย่างกัน หากฉันต้องการค้นหาและเลิกใช้งานมันในที่เก็บทั้งหมดของเรา: การ ค้นหา`lodash.map`จะส่งกลับผลลัพธ์ 0 รายการ การค้นหา`lodash`จะส่งคืนอินสแตนซ์ lodash ทั้งหมด (กว้างเกินไป) การค้นหา`map`จะส่งคืน`lodash.map`อินสแตนซ์และ`[array].map`อินสแตนซ์ (กว้างเกินไป)
> 
> ในการทำในเครื่องพีซีของฉัน ฉันจะต้องโคลนที่เก็บหลายร้อยแห่ง จากนั้นค้นหาในโฟลเดอร์ที่มีที่เก็บเหล่านั้น
> 
> มันจะเป็นไปได้ที่จะเพิ่มคุณลักษณะนี้ไปอยู่ที่ใหม่**ค้นหารหัส** ? ขอขอบคุณ

Max

**Comment by ghost** [comment]
> ในองค์กรที่มีที่เก็บหลายร้อยแห่ง ไม่มีทาง (ที่ฉันรู้) ที่จะค้นหาโดยใช้`.`. เครื่องหมายจุดมีความสำคัญเป็นพิเศษเนื่องจากใช้เพื่อเข้าถึงคุณสมบัติ วิธีการ ฯลฯ ในภาษาการเขียนโปรแกรมหลายภาษา
> 
> มาดู`lodash.map`ตัวอย่างกัน หากฉันต้องการค้นหาและเลิกใช้งานมันในที่เก็บทั้งหมดของเรา: การ ค้นหา`lodash.map`จะส่งกลับผลลัพธ์ 0 รายการ การค้นหา`lodash`จะส่งคืนอินสแตนซ์ lodash ทั้งหมด (กว้างเกินไป) การค้นหา`map`จะส่งคืน`lodash.map`อินสแตนซ์และ`[array].map`อินสแตนซ์ (กว้างเกินไป)
> 
> ในการทำในเครื่องพีซีของฉัน ฉันจะต้องโคลนที่เก็บหลายร้อยแห่ง จากนั้นค้นหาในโฟลเดอร์ที่มีที่เก็บเหล่านั้น
> 
> มันจะเป็นไปได้ที่จะเพิ่มคุณลักษณะนี้ไปอยู่ที่ใหม่**ค้นหารหัส** ? ขอขอบคุณ

Max

**Comment by mrsarthakgupta** [comment]
Right now the dot does make things tricky because it’s treated as a separator rather than part of an identifier in many search tools, including classic GitHub search.

With the new code search, there are a couple of partial workarounds, but it still isn’t perfect:

You can scope across all repos in an org using something like:
org:your-org "lodash.map"
(the quotes help force a literal string match instead of splitting on the dot)

If you have the new code search enabled and regex support in your environment, patterns like lodash\.map can sometimes work better than plain text.

That said, your use case (org-wide “find and deprecate all uses of X.y”) is exactly the kind of thing code search is meant to help with, and having first-class support for dot notation / member access as a single token would be really valuable.

So +1 from me — being able to literally search for lodash.map across all repositories in an organization, without cloning everything locally, would make refactors like this much more practical.
